ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -307,6 +307,10 @@ private void recover(MessageStoreRecovery recovery) throws StoreException, IOExc
               info.getOperationTimeMs());
           logger.info("Index : {} updated message with key {} by inserting delete entry of size {} ttl {}", dataDir,
               info.getStoreKey(), info.getSize(), info.getExpirationTimeInMs());
+        } else if (info.isTtlUpdated()) {
+          markAsPermanent(info.getStoreKey(), new FileSpan(runningOffset, infoEndOffset), info.getOperationTimeMs());
```

Comment:
Question: it's adding a ttl update record, why it is called`markAsPermanent()` ?

Reply:
Right now, the only supported ttl update is to make the blob permanent - quite a few assumptions in the code about this.

Reply:
Hmm - that's true. Let's chat to see what we can do. I want to keep it open for upgrades later but yet communicate that only permanency is possible right now